### DISCUSSION TITLE
RootElement

### DISCUSSION THREAD

**Discussion by alrope** [viniciussanchez/RESTRequest4Delphi#148] [Ideas]
Olá, implementei a função RootElement para popular o DataSet com o elemento específico, como um Array por exemplo.

ficou assim: 

 resp := TRequest.New.BaseURL(BASE_URL)
      .Resource('/members')
     .Accept(REST.Types.CONTENTTYPE_APPLICATION_JSON)
     .Adapters(TDataSetSerializeAdapter.New(FDMemTable2))
     .RootElement('lista')
     .Get   ;

 Achei muito útil e prático no meu projeto, caso tenham interesse posso disponibilizar para análise e implementação na futura versão.

**Comment by viniciussanchez** [comment]
Bom dia... seria algo bem interessante sim, mas eu vejo que essa funcionalidade teria que estar dentro do providers, para ficar mais coerente. 
Por exemplo, nessa linha aqui: `.Adapters(TDataSetSerializeAdapter.New(FDMemTable2))` poderia ficar assim `.Adapters(TDataSetSerializeAdapter.New(FDMemTable2, 'lista'))`.
Eu digo isso porque a criação dos adapters foi justamente tirar o acoplamento do RESTRequest4Delphi com DataSet-Serialize

**Reply by alrope** [reply]
Olá Vinicius, você tem razão, até porque o Adapters pode receber um TArray<IRequestAdapter>, vou tentar implementar sua sugestão para ver como ficaria. Um abraço e parabéns pelos componentes.

**Reply by mariolandossa** [reply]
Boa tarde, estou com esse mesmo problema, isso já foi implementado ?
Obrigado e excelente trabalho

**Reply by viniciussanchez** [reply]
Não foi... podem enviar um pull request se quiserem colaborar com o projeto
